utils/config-processor/CMakeLists.txt

```diff
@@ -1,2 +1,6 @@
 clickhouse_add_executable (config-processor config-processor.cpp)
-target_link_libraries(config-processor PRIVATE clickhouse_common_config_no_zookeeper_log)
+if (ENABLE_SSL)
```

Comment:
The different linking in case of ENABLE_SSL = ON/OFF is quite strange. I don't really get why it is necessary?

One of my original comments was whether we need the encryption/decryption binary at all. Does Linux not offer some command that can do the same job?

Reply:
Done. I removed the conditional linking. It does not make sense. 

> One of my original comments was whether we need the encryption/decryption binary at all. Does Linux not offer some command that can do the same job?

Answered [here](https://github.com/ClickHouse/ClickHouse/pull/50986#discussion_r1269412597).